src/routes/snippets.rs

```diff
@@ -77,6 +77,27 @@ pub fn create_snippet(
     create_snippet_impl(&**storage, &snippet, origin.path())
 }
 
+#[get("/snippets?<title>&<syntax>&<tag>&<marker>&<limit>")]
+pub fn list_snippets(
+    storage: State<Box<dyn Storage>>,
+    title: Option<String>,
+    syntax: Option<String>,
+    tag: Option<String>,
+    marker: Option<String>,
+    limit: Option<usize>,
+    _content_type: NegotiatedContentType,
+    _user: BearerAuth,
+) -> Result<Output<Vec<Snippet>>, ApiError> {
+    let mut criteria = ListSnippetsQuery::default();
+    criteria.title = title;
+    criteria.syntax = syntax;
+    criteria.tags = tag.map(|v| vec![v]);
```

Comment:
**n.b.** I don't know how to extract multiple tags from query string. Taking into account that our python implementation always dealt with a single tag, I think it shouldn't be a problem for initial implementation of this route.